inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -362,7 +362,7 @@ def execute_from_manifest(self,
                 if self._is_ota_no_update_available(cmd_list) and self._package_list == "":
                     # if no package upgrade/install, set the status to OTA_NO_UPDATE and skip saving the granular data.
                     self._update_logger.status = OTA_NO_UPDATE
-                else:
+                elif self.sota_mode != 'download-only':
```

Comment:
the logic is a bit hard to follow for someone reading the code without the context of this bug--please add comment to explain here and also further down (`if self.sota_mode != 'download-only':`) what's happening

Reply:
Comment added.